File: tests-playwright/pages/checkout-complete-page.ts
import { Page, Locator } from '@playwright/test';
import { BasePage } from './base-page';

export class CheckoutCompletePage extends BasePage {
  private readonly completeContainer: Locator;
  private readonly completeHeader: Locator;
  private readonly completeText: Locator;
  private readonly backHomeButton: Locator;

  constructor(page: Page) {
    super(page);
    this.completeContainer = page.locator('#checkout_complete_container');
    this.completeHeader = page.locator('.complete-header');
    this.completeText = page.locator('.complete-text');
    this.backHomeButton = page.locator('[data-test="back-to-products"]');
  }

  async getCompleteHeader(): Promise<string> {
    return await this.getElementText(this.completeHeader);
  }

  async getCompleteText(): Promise<string> {
    return await this.getElementText(this.completeText);
  }

  async backToProducts(): Promise<void> {
    await this.clickElement(this.backHomeButton);
  }

  async isCompletePageLoaded(): Promise<boolean> {
    return await this.isElementVisible(this.completeContainer);
  }
}

File: tests-csharp/Pages/CheckoutCompletePage.cs
using Microsoft.Playwright;
using Tests.CSharp.Pages;

namespace Tests.CSharp.Pages;

public class CheckoutCompletePage : BasePage
{
    private readonly ILocator _completeHeader;
    private readonly ILocator _completeText;
    private readonly ILocator _backHomeButton;

    public CheckoutCompletePage(IPage page) : base(page)
    {
        _completeHeader = Page.Locator(".complete-header");
        _completeText = Page.Locator(".complete-text");
        _backHomeButton = Page.Locator("[data-test='back-to-products']");
    }

    public async Task<string> GetCompleteHeaderAsync()
    {
        return await GetElementTextAsync(_completeHeader);
    }

    public async Task<string> GetCompleteTextAsync()
    {
        return await GetElementTextAsync(_completeText);
    }

    public async Task BackHomeAsync()
    {
        await ClickElementAsync(_backHomeButton);
    }
}

